protos/flyteidl/core/tasks.proto

```diff
@@ -153,5 +153,47 @@ message Container {
 
     // Ports to open in the container. This feature is not supported by all execution engines. (e.g. supported on K8s but
     // not supported on AWS Batch)
+    // Only K8s
     repeated ContainerPort ports = 7;
+
+    // BETA: Optional configuration for DataLoading. If not specified, then default values are used.
+    // This makes it possible to to run a completely portable container, that uses inputs and outputs
+    // only from the local file-system and without having any reference to flyteidl. This is supported only on K8s at the moment.
+    // If data loading is enabled, then data will be mounted in accompanying directories specified in the DataLoadingConfig. If the directories 
+    // are not specified, inputs will be mounted onto and outputs will be uploaded from a pre-determined file-system path. Refer to the documentation
+    // to understand the default paths.
+    // Only K8s
+    DataLoadingConfig data_config = 9;
+}
+
+// This configuration allows executing raw containers in Flyte using the Flyte CoPilot system.
+// Flyte CoPilot, eliminates the needs of flytekit or sdk inside the container. Any inputs required by the users container are side-loaded in the input_path
+// Any outputs generated by the user container - within output_path are automatically uploaded.
+message DataLoadingConfig {
+  // MetadataFormat decides the encoding format in which the input metadata should be made available to the containers. 
```

Comment:
Sorry to be nitpicky, but I've read these comments three times now and I'm still confused.  What is `MetadataFormat`?  The comment says "... format in which the input metadata...". What is input metadata and how is it different than input?  It sounds like it's the same thing.  If it's an input, can we not just call it "Input"?  Currently the situation makes it sound like: 1) there's input data; 2) there's metadata about that input; 3) the format of that metadata can be configured; but 4) the format of the input itself cannot be configured.

Sorry to be annoying but this is the IDL - names here stick around forever.

Reply:
i think we just need to write better documentation about metadata. as this is not really data in all cases.